File: GuildWarsPartySearch/Models/Endpoints/PartySearchEntry.cs
using Newtonsoft.Json;

namespace GuildWarsPartySearch.Server.Models.Endpoints;

public sealed class PartySearchEntry
{
    [JsonProperty(nameof(PartySize))]
    public int? PartySize { get; set; }

    [JsonProperty(nameof(PartyMaxSize))]
    public int? PartyMaxSize { get; set; }

    [JsonProperty(nameof(Npcs))]
    public int? Npcs { get; set; }
}

File: GuildWarsPartySearch.BotUploader/source/payloads/PartySearchEntry.cpp
#include <cstdint>
#include <json.hpp>
#include <payloads/PartySearchEntry.h>

using json = nlohmann::json;

namespace Payloads {
    void to_json(json& j, const PartySearchEntry& p) {
        j["PartySize"] = p.PartySize;
        j["PartyMaxSize"] = p.PartyMaxSize;
        j["Npcs"] = p.Npcs;
        j["CharName"] = p.CharName;
    }
}